Feature/HM-54-[Auth]-Use Preferences

petarbelchev opened the pull request:
The preferences functionality for the buyer has been implemented."

petarbelchev committed:
The preferences functionality for the buyer has been implemented.

rosteslav commented on src/Auth/Auth.Api/Controllers/AuthenticateController.cs:
```diff
@@ -70,6 +70,40 @@ public async Task<IActionResult> Register([FromBody] RegisterUserModel model)
+        [ProducesResponseType(typeof(Response), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(Response), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(Response), StatusCodes.Status409Conflict)]
+        public async Task<IActionResult> RegisterBuyer([FromBody] RegisterBuyerModel model)
```
from task: "Preferences should act like additionalInfo for Sellers and broker but for Buyers. The values should be from preferences entity. Should be accepted on register form."
No new endpoint, fields are added in the register! how does AdditionalInfoWork?

rosteslav replied:
In `AuthOptionsController`, you should add, however, everything that is needed for setting the value for preferences. 
this should include:
-BuildingType
-Purpose 
-Neighborhood.Region

petarbelchev committed:
The POST /register/buyer endpoint has been removed and the logic has been moved into the POST /register endpoint.

petarbelchev committed:
A new GET /AuthOptions/preferences endpoint has been added.
PreferencesRepository has been removed and his logic has been moved into the AuthOptionsRepository.

rosteslav committed:
- get latest on dev
